pkg/plugins/resources/shell/environments.go

```diff
@@ -0,0 +1,70 @@
+package shell
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+type Environments []Environment
+
+func (e *Environments) ToStringArray() []string {
+	result := []string{}
+
+	for _, environment := range *e {
+		result = append(result, environment.String())
+	}
```

Comment:
```suggestion
	result := make([]string, len(*e))

	for i, environment := range *e {
		result[i] = environment.String()
	}
```

Nit: you can pre allocate memory when doing this kind of operation, this saves some costly array copies when calling `append` on a full array. That's performance optimization thought.